src/datomish/exec.cljc

```diff
@@ -0,0 +1,67 @@
+(ns datomish.exec
+  #?(:cljs
+     (:require-macros
+      [datomish.pair-chan :refer [go-pair <?]]
+      [cljs.core.async.macros :refer [go]]))
+  (:require
+     [datomish.sqlite :as s]
+     [datomish.sqlite-schema :as ss]
+     [datomish.query :as dq]
+     #?@(:clj
+           [[datomish.jdbc-sqlite :as j]
+            [datomish.pair-chan :refer [go-pair <?]]
+            [clojure.core.async :refer [go <!! <! >! chan close! take!]]])
+     #?@(:cljs
+           [[datomish.promise-sqlite :as j]
+            [datomish.pair-chan]
+            [cljs.core.async :as a :refer [<!! <! >! chan close! take!]]])))
+
+(defn <run
```

Comment:
I think results will *ultimately* end up collected in memory; we won't hand a channel back to JS consumers, and we need a bounded lifetime. There are good reasons to work with streams within that:

- Capped memory use, modulo buffering. The current implementation should block the sqlite producer once the buffer is full, keeping the cursor open until exhausted; this channel is a lot like getting a cursor instance in a traditional sqlite library. IMO that's better than slurping the entire cursor into in-memory maps.
- Transducing a stream is a good model of what we're going to have to do to implement `pull` in a single query: roll up sequential cursor results from a `GROUP BY`, eventually emitting a single compound result. The row transducer is currently simple, but will eventually map M rows to N results in this way.
- It's a foundation of making limit queries work efficiently, when those queries depend on a join or a condition outside of SQL: the storage cursor needs to be ready to return lots of results, but we probably won't need them all.